### ISSUE TITLE
Caja started to do hyphenation on long file names

### ISSUE THREAD

**Issue by piotrjurkiewicz** [mate-desktop/caja#1284]
Caja started to insert hyphens in long file names. See attached screenshot. It is probably related to the recent "enhancements" in Pango:

https://blogs.gnome.org/mclasen/2019/07/27/more-text-rendering-updates/

https://blogs.gnome.org/mclasen/2019/08/07/pango-1-44-wrap-up/

> As a consequence, there is now a text attribute to suppress the insertion of hyphens.

Maybe this attribute should be enabled in Caja?

![image](https://user-images.githubusercontent.com/2237894/62826218-eeb8ab80-bbb7-11e9-9474-06b139747b1c.png)

#### MATE general version

1.22

#### Package version

community/caja 1.22.1-1
extra/pango 1:1.44.3-1
extra/pangomm 2.42.0-1

#### Linux Distribution

Arch

**Comment by baldr14** [comment]
Here is the Bug-Report for pango: https://gitlab.gnome.org/GNOME/pango/issues/401
The bug seems to be caja related. The bug still exists.


Distribution: Arch Linux
Desktop: gnome 3.34

Packages:
caja 1.22.2
pango 1.44.7
pangomm 2.42.0

**Comment by raveit65** [comment]
Confirmed with fedora 31.
Ihmo, hyphens produce bad grammar ;)
Folder `AndroidStudioProjects` looks like this in caja with normal icon-size and HiDIPI resolution.

```
AndroidStudioProje-
cts
```
I am not a native English speaker but is this correct English grammar?

**Comment by arteme** [comment]
As far as pango is concerned, it's not a bug, but a feature. Other projects had dealt with it:
https://gitlab.gnome.org/GNOME/nautilus/merge_requests/482/diffs

**Comment by sc0w** [comment]
I think it is a crazy new feature with no sense

It was fixed in nautilus with https://gitlab.gnome.org/GNOME/nautilus/commit/9738d8561ed38a87b4d72db3e49653641edd9a09

**Comment by raveit65** [comment]
> As far as pango is concerned, it's not a bug, but a feature. 

Who said this is a bug?

Breaking `AndroidStudioProjects` wrong is in grammar error.

**Comment by raveit65** [comment]
@sc0w 
I agree with adding nautilus fix to caja.

**Comment by baldr14** [comment]
> Who said this is a bug?

I said, but have correct it in my comment ;-)

**Comment by raveit65** [comment]
I got an interesting answer from M.Clasen, looks like we have more option to fix that.
https://gitlab.gnome.org/GNOME/pango/issues/401#note_636509
```
.......you can now influence the result with some attributes:

- allow_breaks : 'true' or 'false' whether to allow line breaks or not. If not allowed, the range will be kept in a single run as far as possible. Breaks are allowed by default.

- insert_hyphens : 'true' or 'false' whether to insert hyphens when breaking lines in the middle of a word. Hyphens are inserted by default.
```

**Comment by raveit65** [comment]
This should fix the issue. https://github.com/mate-desktop/caja/pull/1320
Please test.

**Comment by sc0w** [comment]
I have this issue with caja 1.24.0 (without patches) and pango 1.44.7 in debian testing

![2020-05-26_02-56](https://user-images.githubusercontent.com/7734191/82850445-b68f1900-9efc-11ea-8119-f2eb3b11be16.png)

@monsta @lukefromdc please, can you confirm?

**Comment by lukefromdc** [comment]
With Debian Unstable now shipping libpango-1.0-0_1.44.7-4_amd64.deb and with caja 1.24 from git master as of 5-23-2020 I did NOT get this hypenation on a test file duplicating the exact filename. A second test with Menta for the GTK theme and Mate for the icon theme didn't do this either, even after playing with the filename to put the break in the same place. You might have a condition requiring a specific font width and icon size, plus the repeated periods in the filename all together creating a corner case.

All the commits to caja with "pango" or "hyphen" in the commit description predate the release of 1.24.0 so no change to caja should explaion why you still saw this. That said, this change to pango has been an overall nuisance in many places and in my opinion pango devs should change the internal default for this behavior to FALSE. Just reverting the pango commits would break existing builds of many packages that depend on pango so that would not be a good idea on the part of pango devs.

**Comment by sc0w** [comment]
it is weird, it works fine here building from git locally with https://github.com/mate-desktop/caja/commit/82cd74f8f1db24d2a889770fa35a6866a8e5c694 as latest commit, but I see the issue with official 1.24 deb packages

@lukefromdc can you try with official 1.24 deb packages from debian ?

**Comment by lukefromdc** [comment]
Not very easily, as my packages are organized differently than the way
Debian packages MATE. Mine are one .deb file per source tarball,
Debian splits them up and dependencies may be differently named. A
forced install is thus necessary and breaks the package management
until reverted.

On 5/26/2020 at 6:39 AM, "ZenWalker"  wrote:
	it is weird, it works fine here building from git locally with
82cd74f as latest commit, but I see the issue with official 1.24 deb
packages 

	@lukefromdc can you try with official 1.24 deb packages from debian ?

**Comment by lucas-oliver-trondsen** [comment]
Please, turn off automatic hyphenation of icon names because it's extremely annoying. Many icon names already contain hyphens, so, if additional hyphenes are inserted automatically at the end of lines, I don't know which hyphenes are really part of the names and which are inserted to split long names on multiple lines. I'm using Caja 1.24.0 on Debian 11.

**Comment by lukefromdc** [comment]
Note that with libpango 1.48 and caja 1.26 (on Debian Unstable with locally compiled MATE from git master) I am not getting this problem, testing tonight gave my hyphen-free two line filenames. Wondering if something got fixed in libpango 1.48 after causing problems in 1.46?
